noxfile.py

```diff
@@ -1,41 +1,92 @@
+import sys
+
 import nox
 
-nox.options.sessions = ["tests"]
-python_versions = ("3.10", "3.11", "3.12")
+nox.options.default_venv_backend = "uv"
 
+supported_python_versions = ("3.11", "3.12", "3.13")
 
-@nox.session
-def tests(session):
-    """
-    Run tests with pytest.
-    """
-    pytest_options = {}
-    session.install(".[tests]")
-    session.run("pytest", *pytest_options)
+maxpython = max(supported_python_versions)
+minpython = min(supported_python_versions)
+docpython = "3.13"
+
+current_python = f"{sys.version_info.major}.{sys.version_info.minor}"
+
+nox.options.sessions = [f"tests-{current_python}(all)"]
+
+pytest_command: tuple[str, ...] = (
+    "pytest",
+    "--pyargs",
+    "--durations=5",
+    "--tb=short",
+    "-n=auto",
+    "--dist=loadfile",
+)
+
+with_coverage: tuple[str, ...] = (
+    "--cov=plasmapy",
+    "--cov-report=xml",
+    "--cov-config=pyproject.toml",
+    "--cov-append",
+    "--cov-report",
+    "xml:coverage.xml",
+)
+
+test_specifiers: list = [
+    nox.param("run all tests", id="all"),
+    nox.param("with code coverage", id="cov"),
+    nox.param("lowest-direct", id="lowest-direct"),
+]
+
+
+@nox.session(python=supported_python_versions)
+@nox.parametrize("test_specifier", test_specifiers)
```

Comment:
One of the awesome things about Nox is that we can parametrize testing environments, which was designed to be similar to how `@pytest.mark.parametrize` works.